docs/internals/task_lifecycle.md

```diff
@@ -0,0 +1,149 @@
+---
+title:  "Task Lifecycle"
+nav-title: Task Lifecycle
+nav-parent_id: internals
+nav-pos: 5
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+A task in Flink is the basic unit of execution. This is the place where each parallel instance of your operators gets executed. 
+
+This document goes through the different phases in the lifecycle of the `StreamTask`, which is the base for all different task sub-types in Flink's streaming engine. 
+
+* This will be replaced by the TOC
+{:toc}
+
+## Operator Lifecycle in a nutshell
+
+As the task is the entity that executes your operators, its lifecycle is tightly integrated with that of an operator. Given this, it is worth spending a few lines 
+simply mentioning the basic methods representing the lifecycle of an operator before diving into those of the `StreamTask` itself. The list is presented below in 
+the order that they are called. These are:
+
+        // initialization
+	    initializeState()
+	    setup()
+	    open()
+	
+	    // processing
+	    processElement()
+	    processWatermark()
+	
+	    // termination
+	    close()
+	    dispose()
+
+        // checkpointing
```

Comment:
Move checkpointing into a separate box to make clear it is not included in the regular thread?